### PR TITLE
Bugfix/13847-datalabels-in-navigator

### LINKED ISSUE #13847: Export Highcharts Gantt Chart Error
#### Expected behaviour
Expected to download the chart exactly as developed in the UI like in the picture below
image

#### Actual behaviour
What I actually get to see in jpeg when I download
gantt-chart-with-navigat (1)

#### Live demo with steps to reproduce
https://jsfiddle.net/cvinhar/scx6f4ph/21/

#### Product version
Highcharts Gantt JS v8.1.2 (2020-06-16)
Highcharts JS v8.1.2 (2020-06-16)

#### Affected browser(s)
So far tested in Google Chrome

### PR DESCRIPTION
**Pull request by pawellysy**
Fixed #13847, DataLabels in navigator should be disabled in the default configuration.

### CODE REVIEW

File: ts/Core/Navigator.ts [hunk 1]
```diff
@@ -2241,6 +2241,8 @@ class Navigator {
 
                 baseOptions = base.options || {};
                 baseNavigatorOptions = baseOptions.navigatorOptions || {};
+
+                userNavOptions.dataLabels = splat(userNavOptions.dataLabels);
```
Comment (karolkolodziej) [inline ts/Core/Navigator.ts:2245]: Please add a comment with the issue number here.

File: samples/unit-tests/navigator/navigator/demo.js [hunk 2]
```diff
@@ -1144,3 +1144,58 @@ QUnit.test(
         );
     }
 );
+
+QUnit.test('Navigator dafault dataLabels enabled, #13847.', function (assert) {
+    const chart = Highcharts.stockChart('container', {
+        plotOptions: {
```
Comment (karolkolodziej) [inline samples/unit-tests/navigator/navigator/demo.js:1150]: If you have only one series there is no need to declare the `plotOptions`. Simply add that config in the `series`.

File: samples/unit-tests/navigator/navigator/demo.js [hunk 3]
```diff
@@ -1144,3 +1144,58 @@ QUnit.test(
...
+        chart.navigator.series[0].options.dataLabels[0].enabled,
+        true,
+        'DataLabels in Navigator should be enabled, if specified in options.'
+    );
+
+    chart.update({
+        navigator: {
+            series: {
+                dataLabels: [{
+                    enabled: true
```
Comment (karolkolodziej) [inline samples/unit-tests/navigator/navigator/demo.js:1190]: I would change here to `false`.
Now it doesn't change anything in the chart config, so it's hard to test if this update is performed properly.

File: samples/unit-tests/navigator/navigator/demo.js [hunk 4]
```diff
@@ -1144,3 +1144,58 @@ QUnit.test(
...
+                }]
+            }
+        }
+    });
+
+    assert.equal(
+        chart.navigator.series[0].options.dataLabels[0].enabled,
+        true,
```
Comment (karolkolodziej) [inline samples/unit-tests/navigator/navigator/demo.js:1198]: `false` as a consequence of the comment above

**Review by karolkolodziej — CHANGES_REQUESTED** [review]

**Review by karolkolodziej — APPROVED** [review]
Greate!

File: ts/Core/Navigator.ts [hunk 5]
```diff
@@ -2241,6 +2241,10 @@ class Navigator {
 
                 baseOptions = base.options || {};
                 baseNavigatorOptions = baseOptions.navigatorOptions || {};
+
+                // The dataLabels options are not merged correctly
+                // if the settings are and array, #13847.
```
Comment (raf18seb) [inline ts/Core/Navigator.ts:2246]: Typo "and" here. Maybe
```The dataLabels config is not merged correctly when provided in an array, #13847.```

**Review by raf18seb — APPROVED** [review]
One small typo. Except that everything looks great!

**Review by pawelfus — APPROVED** [review]
Nice!

**Review by TorsteinHonsi — APPROVED** [review]

**Pull request merged**